src/model/Board.java

```diff
@@ -89,6 +119,18 @@ public Square getSquare(int position) {
         return squares;
     }
 
+    public Set<Square> getSquaresForPlayer(PieceColor color) {
+        Set<Square> playersSquares = new HashSet<>();
```

Comment:
Awesome! I definitely agree with all your reasoning. Sounds great to me. :) Thanks for the explanation!